spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/query/MongoQueryCreator.java

```diff
@@ -15,6 +15,7 @@
  */
 package org.springframework.data.mongodb.repository.query;
 
+import static java.util.Objects.nonNull;
```

Comment:
We need to preserve JDK 6 compatibility. Can we just exchange this against a simple null check?